app/src/pages/movies/movie/components/EditMovieForm.jsx

```diff
@@ -27,43 +29,48 @@ const ButtonContainer = styled.div`
 `
 
 function EditMovieForm(props) {  
-    const [genres, setGenres] = useState([]);
-    
-    useEffect(() => {
-        Promise
-            .resolve(genresFromJson)
-            .then(genres => setGenres(genres))
-    }, [])
+    const [title, setTitle] = useState(props.movie.title);
+    const [release_date, setReleaseDate] = useState(props.movie.release_date);
+    const [poster_path, setUrl] = useState(props.movie.poster_path);
+    const [genres, setGenres] = useState(props.movie.genres.join(','));
+    const [overview, setOverview] = useState(props.movie.overview);
+    const [runtime, setRuntime] = useState(props.movie.runtime);
+
+    const handleEditTitle = useCallback(event => setTitle(event.target.value), []);
+    const handleEditReleaseDate = useCallback(event => setReleaseDate(event.target.value), []);
+    const handleEditUrl = useCallback(event => setUrl(event.target.value), []);
+    const handleEditGenres = useCallback(event => setGenres(event.target.value), []);
+    const handleEditOverview = useCallback(event => setOverview(event.target.value), []);
+    const handleEditRuntime = useCallback(event => setRuntime(event.target.value), []);
+
+    function confirmUpdate(){
+        let movie = new Movie(title, release_date, poster_path, genres.split(','), overview, Number.parseInt(runtime));
+        movie.id = props.movie.id;

```

Comment:
u can add id as inner key of the Movie model => and do not mutate data

Reply:
Due to I use one model `Movie` for **add** and for **update** actions, if I add `id` field to `Movie`, I'll get exception 'id is not allowed' during **add** action even if id=null. 
I could create 2 different models for **add** and another for **update** action. Do you think it would be better?
![image](https://user-images.githubusercontent.com/22270928/116112337-233c0600-a6c0-11eb-80bd-dda761e7c563.png)
![image](https://user-images.githubusercontent.com/22270928/116112357-28995080-a6c0-11eb-9e03-1ded3d94f5de.png)